crates/rpc/rpc-eth-types/src/cache/mod.rs

```diff
@@ -65,6 +65,50 @@ type HeaderLruCache<H, L> = MultiConsumerLruCache<B256, H, L, HeaderResponseSend
 pub struct EthStateCache<B: Block, R> {
     to_service: UnboundedSender<CacheAction<B, R>>,
 }
+/// Drop aware sender struct
+#[derive(Debug)]
+pub struct ActionSender<B: Block, R: Send + Sync> {
```

Comment:
we can keep this private

```suggestion
struct ActionSender<B: Block, R: Send + Sync> {
```